front/app/components/AssignBudgetControl/index.tsx

```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
   ideaId: string;
   className?: string;
 }
 
-interface InnerProps extends OuterProps {
-  authUser: GetAuthUserChildProps;
-  idea: GetIdeaChildProps;
-  project: GetProjectChildProps;
-  phases: GetPhasesChildProps;
-  participationContext: IProjectData | IPhaseData;
-  participationContextId: string;
-  participationContextType: IParticipationContextType;
-}
-
 const AssignBudgetControl = memo(
-  ({
-    view,
-    ideaId,
-    authUser,
-    idea,
-    phases,
-    participationContext,
-    participationContextId,
-    participationContextType,
-    className,
-  }: InnerProps) => {
+  ({ view, ideaId, className, projectId }: Props) => {
+    const authUser = useAuthUser();
+    const idea = useIdea({ ideaId });
+    const project = useProject({ projectId });
+    const phases = usePhases(projectId);
+    const isContinuousProject =
+      project?.attributes.process_type === 'continuous';
+    const ideaPhaseIds = !isNilOrError(idea)
+      ? idea?.relationships?.phases?.data?.map((item) => item.id)
+      : null;
+    const ideaPhases = !isNilOrError(phases)
+      ? phases?.filter((phase) => includes(ideaPhaseIds, phase.id))
+      : null;
+    const latestRelevantIdeaPhase = ideaPhases
+      ? getLatestRelevantPhase(ideaPhases)
+      : null;
+    const participationContext = isContinuousProject
+      ? project
+      : latestRelevantIdeaPhase;
+    const participationContextType = isContinuousProject ? 'project' : 'phase';
+    const participationContextId = participationContext?.id || null;
     const basket = useBasket(
       participationContext?.relationships?.user_basket?.data?.id
     );
 
     const [processing, setProcessing] = useState(false);
 
-    const handleAddRemoveButtonClick = (event?: FormEvent) => {
-      event?.preventDefault();
-      assignBudget();
-    };
-
-    const assignBudget = async () => {
-      const timeout = (ms: number) =>
-        new Promise((resolve) => setTimeout(resolve, ms));
-      const done = async () => {
-        await timeout(200);
-        setProcessing(false);
-      };
+    const handleAddRemoveButtonClick =
+      (idea: IIdeaData, participationContextId: string) =>
+      (event?: FormEvent) => {
+        event?.preventDefault();
 
-      if (!isNilOrError(idea)) {
         const isBudgetingEnabled =
           idea.attributes.action_descriptor.budgeting?.enabled;
-        const basketIdeaIds = !isNilOrError(basket)
-          ? basket.relationships.ideas.data.map((idea) => idea.id)
-          : [];
-        const isInBasket = includes(basketIdeaIds, ideaId);
         const budgetingDisabledReason =
           idea.attributes.action_descriptor.budgeting?.disabled_reason;
 
         if (
+          // not signed up/in
           isNilOrError(authUser) ||
-          budgetingDisabledReason === 'not_verified'
+          budgetingDisabledReason === 'not_signed_in'
         ) {
           openSignUpInModal({
-            verification: budgetingDisabledReason === 'not_verified',
```

Comment:
verification doesn't work here. For one because this disabled reason is always false here, but when I set it manually to true, it didn't work either. I'll log this as an extra bug.

Reply:
Definitely should check this!